client/app/bundles/course/assessment/pages/AssessmentStatistics/AnswerDisplay/AllAttemptsDisplay.tsx

```diff
@@ -63,63 +110,113 @@ const AllAttemptsDisplay: FC<Props> = (props) => {
+    if (answerStatus === 'completed') {
+      return (
+        <>
+          <Typography variant="body1">
+            {t(translations.pastAnswerTitle, {
+              submittedAt: formatLongDateTime(answerDetails!.createdAt),
+            })}
+          </Typography>
+          <AnswerDetails answer={answerDetails!} question={question!} />
+        </>
+      );
+    }
+    if (answerStatus === 'errored') {
+      return (
+        <Alert severity="error">{t(messagesTranslations.fetchingError)}</Alert>
+      );
+    }
+    if (answerStatus === 'pending') {
+      return <LoadingIndicator />;
+    }
+    return null;
+  };
+
+  const changeDisplayedIndex = (newIndex: number): void => {
+    if (
+      answersCache[allAnswers[newIndex].id] &&
+      answersCache[allAnswers[newIndex].id].status === 'completed'
+    ) {
+      setSliderIndex(newIndex);
+      setDisplayedIndex(newIndex);
+    } else {
+      tryFetchAnswerById(allAnswers[newIndex].id).finally(() => {
+        setSliderIndex(newIndex);
+        setDisplayedIndex(newIndex);
+      });
+    }
+  };
+
+  return (
+    <>
+      {renderQuestionComponent()}
       {answerSubmittedTimes.length > 1 && (
-        <div className="w-[calc(100%_-_17rem)] mx-auto">
+        <div className="w-[calc(100%_-_17rem)] mx-auto pt-8">
           <CustomSlider
             defaultValue={currentAnswerMarker.value}
             marks={answerSubmittedTimes}
             max={currentAnswerMarker.value}
             min={earliestAnswerMarker.value}
-            onChange={(_, value) => {
-              setDisplayedIndex(Array.isArray(value) ? value[0] : value);
+            onChange={(event, value) => {
+              const newIndex = Array.isArray(value) ? value[0] : value;
+              if (event.type.includes('mouse')) {
+                setSliderIndex(newIndex);
+              } else {
+                changeDisplayedIndex(newIndex);
+              }
+            }}
+            onChangeCommitted={(_, value) => {
+              const newIndex = Array.isArray(value) ? value[0] : value;
+              changeDisplayedIndex(newIndex);
             }}
             step={null}
-            valueLabelDisplay="off"
+            valueLabelDisplay="on"
+            valueLabelFormat={(value) =>
+              `${formatLongDateTime(allAnswers[value].createdAt)} (${value + 1} of ${allAnswers.length})`
+            }
           />
         </div>
       )}
-
-      <Typography variant="h6">
-        {t(translations.pastAnswerTitle, {
-          submittedAt: formatLongDateTime(
-            sortedAnswers[displayedIndex ?? answerSubmittedTimes.length - 1]
-              .createdAt,
-          ),
-        })}
-      </Typography>
-      <AnswerDetails
-        answer={
-          sortedAnswers[displayedIndex ?? answerSubmittedTimes.length - 1]
-        }
-        question={question}
-      />
+      <div className={sliderIndex === displayedIndex ? '' : 'opacity-60'}>
```

Comment:
nit: consider `sliderIndex !== displayedIndex && 'opacity-60'`  
but don't feel particularly strongly about it either way

Reply:
I tried that and it doesn't work on native divs. Perhaps it only works on React components?